### PR TITLE
populate request.postData and response.content.encoding in HarEntry

### PR DESCRIPTION
**Pull request by yysushi**
## Added Features

- request.postData
it would be great to populate request.postData in HarEntry as response.content done.
I picked up only "mimeType" and "text" as sub properties (I mean I skipped "params" and "comment").
- response.content.encoding
I also populate how response content is encoded. it's necessary for users to analyze the actual response content.

### CODE REVIEW

**Comment by yysushi** [comment]
<details>
<summary> make lint & test log in my laptop </summary>

```bash
y-tsuji: ~/g/g/y/haralyzer (req_postdata)$ git log -1 --oneline
───────┬──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
       │ STDIN
───────┼──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
   1   │ 13d3ff2 (HEAD -> req_postdata, origin/req_postdata) populate request.postData in HarEntry
───────┴──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
y-tsuji: ~/g/g/y/haralyzer (req_postdata)$ make lint
black --check haralyzer
All done! ✨ 🍰 ✨
6 files would be left unchanged.
pylint --disable=E1101,C0103,W0511,R0901 haralyzer

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)

flake8 --max-line-length 89 --statistics --show-source --count haralyzer
0
bandit -r haralyzer
[main]  INFO    profile include tests: None
[main]  INFO    profile exclude tests: None
[main]  INFO    cli include tests: None
[main]  INFO    cli exclude tests: None
[main]  INFO    running on Python 3.9.9
Run started:2022-12-27 22:10:24.173767

Test results:
        No issues identified.

Code scanned:
        Total lines of code: 1398
        Total lines skipped (#nosec): 0

Run metrics:
        Total issues (by severity):
                Undefined: 0.0
                Low: 0.0
                Medium: 0.0
                High: 0.0
        Total issues (by confidence):
                Undefined: 0.0
                Low: 0.0
                Medium: 0.0
                High: 0.0
Files skipped (0):
y-tsuji: ~/g/g/y/haralyzer (req_postdata)$ echo $?
0
y-tsuji: ~/g/g/y/haralyzer (req_postdata)$ make test
py.test --cov haralyzer tests/ -vv
======================================================================== test session starts =========================================================================
platform darwin -- Python 3.9.9, pytest-6.2.5, py-1.11.0, pluggy-1.0.0 -- /nix/store/v663g4j00mw5lxk6iyd0zigbv93j2rg2-python3-3.9.9-env/bin/python3.9
cachedir: .pytest_cache
rootdir: /Users/y-tsuji/git/github.com/yysushi/haralyzer
plugins: cov-3.0.0
collected 62 items

tests/test_entry.py::test_entry PASSED                                                                                                                         [  1%]
tests/test_entry.py::test_request PASSED                                                                                                                       [  3%]
tests/test_entry.py::test_response PASSED                                                                                                                      [  4%]
tests/test_entry.py::test_backwards PASSED                                                                                                                     [  6%]
tests/test_multi_parser.py::test_init PASSED                                                                                                                   [  8%]
tests/test_multi_parser.py::test_load_times PASSED                                                                                                             [  9%]
tests/test_multi_parser.py::test_stdev PASSED                                                                                                                  [ 11%]
tests/test_page.py::test_init PASSED                                                                                                                           [ 12%]
tests/test_page.py::test_no_title PASSED                                                                                                                       [ 14%]
tests/test_page.py::test_filter_entries PASSED                                                                                                                 [ 16%]
tests/test_page.py::test_filter_entries_load_time PASSED                                                                                                       [ 17%]
tests/test_page.py::test_get_load_time PASSED                                                                                                                  [ 19%]
tests/test_page.py::test_entries PASSED                                                                                                                        [ 20%]
tests/test_page.py::test_iteration PASSED                                                                                                                      [ 22%]
tests/test_page.py::test_file_types PASSED                                                                                                                     [ 24%]
tests/test_page.py::test_request_types PASSED                                                                                                                  [ 25%]
tests/test_page.py::test_sizes_trans PASSED                                                                                                                    [ 27%]
tests/test_page.py::test_sizes PASSED                                                                                                                          [ 29%]
tests/test_page.py::test_load_times PASSED                                                                                                                     [ 30%]
tests/test_page.py::test_time_to_first_byte PASSED                                                                                                             [ 32%]
tests/test_page.py::test_hostname PASSED                                                                                                                       [ 33%]
tests/test_page.py::test_url PASSED                                                                                                                            [ 35%]
tests/test_page.py::test_duplicate_urls_count PASSED                                                                                                           [ 37%]
tests/test_parser.py::test_init PASSED                                                                                                                         [ 38%]
tests/test_parser.py::test_from_file PASSED                                                                                                                    [ 40%]
tests/test_parser.py::test_from_string PASSED                                                                                                                  [ 41%]
tests/test_parser.py::test_init_entry_with_no_pageref PASSED                                                                                                   [ 43%]
tests/test_parser.py::test_init_entry_with_no_page PASSED                                                                                                      [ 45%]
tests/test_parser.py::test_match_headers PASSED                                                                                                                [ 46%]
tests/test_parser.py::test_match_request_type PASSED                                                                                                           [ 48%]
tests/test_parser.py::test_match_status_code PASSED                                                                                                            [ 50%]
tests/test_parser.py::test_http_version PASSED                                                                                                                 [ 51%]
tests/test_parser.py::test_match_content_type PASSED                                                                                                           [ 53%]
tests/test_parser.py::test_create_asset_timeline PASSED                                                                                                        [ 54%]
tests/chrome/test_chrome_entry.py::test_entry PASSED                                                                                                           [ 56%]
tests/chrome/test_chrome_entry.py::test_request PASSED                                                                                                         [ 58%]
tests/chrome/test_chrome_entry.py::test_request_post PASSED                                                                                                    [ 59%]
tests/chrome/test_chrome_entry.py::test_response PASSED                                                                                                        [ 61%]
tests/chrome/test_chrome_entry.py::test_backwards PASSED                                                                                                       [ 62%]
tests/chrome/test_chrome_page.py::test_init PASSED                                                                                                             [ 64%]
tests/chrome/test_chrome_page.py::test_filter_entries PASSED                                                                                                   [ 66%]
tests/chrome/test_chrome_page.py::test_entries PASSED                                                                                                          [ 67%]
tests/chrome/test_chrome_page.py::test_request_types PASSED                                                                                                    [ 69%]
tests/chrome/test_chrome_page.py::test_load_times PASSED                                                                                                       [ 70%]
tests/chrome/test_chrome_page.py::test_time_to_first_byte PASSED                                                                                               [ 72%]
tests/chrome/test_chrome_page.py::test_hostname PASSED                                                                                                         [ 74%]
tests/chrome/test_chrome_page.py::test_url PASSED                                                                                                              [ 75%]
tests/chrome/test_chrome_page.py::test_redirect PASSED                                                                                                         [ 77%]
tests/firefox/test_firefox_entry.py::test_entry PASSED                                                                                                         [ 79%]
tests/firefox/test_firefox_entry.py::test_request PASSED                                                                                                       [ 80%]
tests/firefox/test_firefox_entry.py::test_request_post PASSED                                                                                                  [ 82%]
tests/firefox/test_firefox_entry.py::test_response PASSED                                                                                                      [ 83%]
tests/firefox/test_firefox_entry.py::test_backwards PASSED                                                                                                     [ 85%]
tests/firefox/test_firefox_page.py::test_init PASSED                                                                                                           [ 87%]
tests/firefox/test_firefox_page.py::test_filter_entries PASSED                                                                                                 [ 88%]
tests/firefox/test_firefox_page.py::test_get_load_time PASSED                                                                                                  [ 90%]
tests/firefox/test_firefox_page.py::test_entries PASSED                                                                                                        [ 91%]
tests/firefox/test_firefox_page.py::test_request_types PASSED                                                                                                  [ 93%]
tests/firefox/test_firefox_page.py::test_load_times PASSED                                                                                                     [ 95%]
tests/firefox/test_firefox_page.py::test_time_to_first_byte PASSED                                                                                             [ 96%]
tests/firefox/test_firefox_page.py::test_hostname PASSED                                                                                                       [ 98%]
tests/firefox/test_firefox_page.py::test_url PASSED                                                                                                            [100%]

---------- coverage: platform darwin, python 3.9.9-final-0 -----------
Name                    Stmts   Miss  Cover
-------------------------------------------
haralyzer/__init__.py       3      0   100%
haralyzer/assets.py       399      4    99%
haralyzer/errors.py         1      0   100%
haralyzer/http.py         107      0   100%
haralyzer/mixins.py        27      0   100%
haralyzer/multihar.py      80      0   100%
-------------------------------------------
TOTAL                     617      4    99%


========================================================================= 62 passed in 9.16s =========================================================================
y-tsuji: ~/g/g/y/haralyzer (req_postdata)$ echo $?
0
```
</details>

File: haralyzer/http.py [hunk 1]
```diff
@@ -250,9 +270,17 @@ def mimeType(self) -> str:
...
 
     @cached_property
-    def text(self) -> str:
+    def text(self) -> Optional[str]:
         """
         :return: Response body
         :rtype: str
         """
-        return self.raw_entry["content"]["text"]
+        return self.raw_entry["content"].get("text")
```
Comment (yysushi) [inline haralyzer/http.py:278]: response.content is required but response.content.text is optional. so, I fixed it too this time.
however, this could appear **breaking** change for users. the actual impact is shown in test_chrome_entry.py::test_response.

Comment (Cyb3r-Jak3) [reply haralyzer/http.py:278]: Is there a reason that this change needs to be made with this PR?

**Comment by Cyb3r-Jak3** [comment]
@yysushi Thanks for the PR!
Can you merge the latest master into your branch? I made some changes which should fix the tests.

File: haralyzer/http.py (hunk 1 above)
Comment (yysushi) [reply haralyzer/http.py:278]: no special reason. will revert it.

**Review by Cyb3r-Jak3 — APPROVED** [review]
LGTM. Thanks again!
